spring-integration-core/src/main/java/org/springframework/integration/dispatcher/BroadcastingDispatcher.java

```diff
@@ -158,13 +160,23 @@ public boolean dispatch(Message<?> message) {
 			throw new MessageDispatchingException(message, "Dispatcher has no subscribers");
 		}
 		int sequenceSize = handlers.size();
+		Message<?> messageToSend = message;
+		UUID sequenceId = null;
+		if (this.applySequence) {
+			sequenceId = message.getHeaders().getId();
+		}
 		for (MessageHandler handler : handlers) {
-			Message<?> messageToSend = message;
 			if (this.applySequence) {
-				messageToSend = getMessageBuilderFactory()
-						.fromMessage(message)
-						.pushSequenceDetails(message.getHeaders().getId(), sequenceNumber++, sequenceSize)
-						.build();
+				if (message instanceof ThreadStatePropagationChannelInterceptor.MessageWithThreadState) {
```

Comment:
Ha! I thought here about minimal, quick, unbreakable, backporting fix.
But I see your point.
Seems for me much cleaner solution and enough flexible.
Let me come up with something :smile: 

Thank you!
And welcome back with fresh mind! :wink: